client/reader/search-stream/site-results.jsx

```diff
@@ -81,14 +81,42 @@ export default connect(
 			sort: ownProps.sort,
 		} );
 
-		// Check if searchResults has one item and if it has a feed_ID
-		if ( searchResults && searchResults.length === 1 ) {
-			let feed = searchResults[ 0 ];
-			if ( feed.feed_ID.length > 0 ) {
-				// If it has a feed_id, get the feed object from the state
-				feed = getFeed( state, feed.feed_ID );
-			}
-			ownProps.onReceiveSearchResults( feed );
+		// Check if searchResults has any feeds
+		if ( searchResults && searchResults.length > 0 ) {
+			const feeds = searchResults;
+			// We want to create a list of unique feeds based on searchResults
+			// We need to do this because the search results may contain duplicate feeds URLs with different http or https schemes
+			// eslint-disable-next-line wpcalypso/redux-no-bound-selectors
+			const feedResults = feeds.reduce( ( uniqueFeeds, feed ) => {
+				// Strip out the URL scheme for subscribe_URL
+				const strippedSubscribeURL = feed.subscribe_URL.replace( /^https?:\/\//, '' );
+
+				// Check if the array already has an item with the same strippedSubscribeURL or feed_ID
+				const foundItem = uniqueFeeds.find(
+					// eslint-disable-next-line wpcalypso/redux-no-bound-selectors
+					( uniqueFeed ) =>
+						( ! feed.feed_ID &&
+							uniqueFeed.subscribe_URL.replace( /^https?:\/\//, '' ) === strippedSubscribeURL ) ||
```

Comment:
I was able to get a NPE because subscribe_URL was null. This was when I searched for "longreads", I believe longreads.com does have both an HTTP and HTTPS version